samples/colab/pytorch_jit.ipynb

```diff
@@ -0,0 +1,440 @@
+            "Requirement already satisfied: nvidia-cusparse-cu12==12.1.0.106 in /usr/local/lib/python3.10/dist-packages (from torch>=2.1.0->shark-turbine==0.9.1.dev1) (12.1.0.106)\n",
+            "Requirement already satisfied: nvidia-nccl-cu12==2.18.1 in /usr/local/lib/python3.10/dist-packages (from torch>=2.1.0->shark-turbine==0.9.1.dev1) (2.18.1)\n",
+            "Requirement already satisfied: nvidia-nvtx-cu12==12.1.105 in /usr/local/lib/python3.10/dist-packages (from torch>=2.1.0->shark-turbine==0.9.1.dev1) (12.1.105)\n",
+            "Requirement already satisfied: triton==2.1.0 in /usr/local/lib/python3.10/dist-packages (from torch>=2.1.0->shark-turbine==0.9.1.dev1) (2.1.0)\n",
+            "Requirement already satisfied: nvidia-nvjitlink-cu12 in /usr/local/lib/python3.10/dist-packages (from nvidia-cusolver-cu12==11.4.5.107->torch>=2.1.0->shark-turbine==0.9.1.dev1) (12.2.140)\n",
+            "Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.10/dist-packages (from jinja2->torch>=2.1.0->shark-turbine==0.9.1.dev1) (2.1.3)\n",
+            "Requirement already satisfied: mpmath>=0.19 in /usr/local/lib/python3.10/dist-packages (from sympy->torch>=2.1.0->shark-turbine==0.9.1.dev1) (1.3.0)\n",
+            "Building wheels for collected packages: shark-turbine\n",
+            "  Building wheel for shark-turbine (pyproject.toml) ... \u001b[?25l\u001b[?25hdone\n",
+            "  Created wheel for shark-turbine: filename=shark_turbine-0.9.1.dev1-py3-none-any.whl size=69686 sha256=1167a861f0c98c2886c4195b05ee118464682392377a65032169394f8d8e87c9\n",
+            "  Stored in directory: /tmp/pip-ephem-wheel-cache-10cp3z6m/wheels/93/49/7b/dd5e1a86a1f80aee91a41ee39b2e8da538043a07c607de4a7c\n",
+            "Successfully built shark-turbine\n",
+            "Installing collected packages: shark-turbine\n",
+            "Successfully installed shark-turbine-0.9.1.dev1\n"
+          ]
+        }
+      ],
+      "source": [
+        "#@title Install SHARK-Turbine following [the instructions](https://github.com/nod-ai/SHARK-Turbine)\n",
+        "\n",
+        "# Limit cell height.\n",
+        "from IPython.display import Javascript\n",
+        "display(Javascript('''google.colab.output.setIframeHeight(0, true, {maxHeight: 300})'''))\n",
+        "\n",
+        "# Note: the install steps are tweaked here to work in Colab.\n",
+        "![[ -d SHARK-Turbine ]] || git clone https://github.com/nod-ai/SHARK-Turbine.git\n",
+        "!python -m pip install --upgrade -r SHARK-Turbine/requirements.txt\n",
+        "!python -m pip install git+https://github.com/nod-ai/SHARK-Turbine.git --upgrade"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "#@title Test installs and report version information\n",
+        "from shark_turbine.dynamo.importer import FxImporter  # Test for install success\n",
```

Comment:
https://github.com/openxla/iree/actions/runs/6462150219/job/17543263981#step:3:309

Looks like Turbine requires Python > 3.10 (https://docs.python.org/3/library/types.html#types.NoneType)
```python
File /work/.notebook.venv/lib/python3.9/site-packages/shark_turbine/dynamo/importer.py:10
      8 import operator
      9 import re
---> 10 from types import NoneType, BuiltinMethodType, BuiltinFunctionType
     11 from typing import Any, Callable, Dict, List, Optional, Sequence, Set, Tuple, Union
     12 import numpy as np

ImportError: cannot import name 'NoneType' from 'types' (/usr/lib/python3.9/types.py)
```

Our Docker image installs 3.9: https://github.com/openxla/iree/blob/77a8741a9eb5a03404bcbb3343f90ff250d11099/build_tools/docker/dockerfiles/base.Dockerfile#L61-L62

I should just update us to use Colab's Dockerfile instead: https://github.com/openxla/iree/issues/15070